NetworthManager.js

```diff
@@ -0,0 +1,207 @@
+const { calculateNetworth, calculateItemNetworth } = require('./helper/calculateNetworth');
+const { parseItems, postParseItems } = require('./helper/parseItems');
+const { NetworthError, PricesError, ItemsError } = require('./helper/errors');
+const NetworthTypes = require('./NetworthTypes');
+const axios = require('axios');
+const { setItems } = require('./constants/itemsMap');
+
+class NetworthManager {
+    /**
+     * Create a new NetworthManager instance
+     * @param {Object} options - Options for the NetworthManager
+     * @param {NetworthTypes} [options.networthType=NetworthTypes.Normal] - The type of networth calculation to use
+     * @param {boolean} [options.cachePrices=true] - Whether to cache the prices for 5 minutes after fetching them or fetch them every time
+     * @param {number} [options.pricesRetries=3] - The amount of retries to fetch the prices when failing to fetch them
+     * @param {number} [options.itemsRetries=3] - The amount of retries to fetch the items when failing to fetch them
+     * @param {number} [options.itemsInterval=1000 * 60 * 60 * 12] - The interval to fetch the items from the Hypixel API
+     * @param {boolean} [options.onlyNetworth=false] - Whether to only return the total networth or the items as well
+     * @param {boolean} [options.stackItems=true] - Whether to stack items with the same name and price
+     * @param {boolean} [options.includeItemData=false] - Whether to include the item data as a property in the item object
+     */
+    constructor({ networthType, cachePrices, pricesRetries, itemsRetries, itemsInterval, onlyNetworth, stackItems, includeItemData }) {
+        this.networthType = networthType || NetworthTypes.Normal;
+        this.cachePrices = cachePrices || true;
+        this.pricesRetries = pricesRetries || 3;
+        this.itemsRetries = itemsRetries || 3;
+        this.itemsInterval = itemsInterval || 1000 * 60 * 60 * 12; // 12 hours
+        this.onlyNetworth = onlyNetworth || false;
+        this.stackItems = stackItems || true;
+        this.includeItemData = includeItemData || false;
+
+        this.itemsPromise = updateItems(this.itemsRetries);
+        setInterval(() => {
+            updateItems(this.itemsRetries);
+        }, this.itemsInterval);
```

Comment:
Ok, ik I just said you don't need setter but this should probably use a setter.

Basically, store the result of calling `setInterval()` to a class variable then when they set `itemsInterval` update the interval.

Ok, I was close. Here's what Chat says:

```js
// Define a variable to hold the interval ID
let intervalId;

// Function to start the interval
function startInterval() {
  intervalId = setInterval(() => {
    console.log('Interval running');
  }, 1000); // Interval time in milliseconds
}

// Function to update the interval
function updateInterval(newIntervalTime) {
  clearInterval(intervalId); // Clear the existing interval
  intervalId = setInterval(() => {
    console.log('Interval running with new interval time');
  }, newIntervalTime); // Set the new interval time
}

// Start the interval
startInterval();

// Update the interval after 5 seconds
setTimeout(() => {
  updateInterval(2000); // Update interval to 2000 milliseconds
}, 5000);
```

Reply:
mh I don't really see the use case of modifying the interval after initializing the manager with the interval

Reply:
Fair, was just thinking for completeness.